pkg/asset/agent/manifests/clusterdeployment.go

```diff
@@ -37,53 +37,53 @@ func (*ClusterDeployment) Name() string {
 func (*ClusterDeployment) Dependencies() []asset.Asset {
 	return []asset.Asset{
 		&agent.OptionalInstallConfig{},
-		// &AgentPullSecret{},
+		&AgentPullSecret{},
```

Comment:
Good point about this causing the ClusterDeployment to get discarded, I forgot dependencies also have that effect :facepalm:

So I guess the best we can do is just generate both with the same name and it's the user's problem if they create them in such a way that the links don't match up.

Reply:
Thanks, fixed. Here is the link for dirty dependencies for ref https://github.com/openshift/installer/blob/master/docs/design/assetgeneration.md#dirty-detection